design/design-doc-composition-functions.md

```diff
@@ -209,570 +738,353 @@ metadata:
   name: example
 spec:
   compositeTypeRef:
-    apiVersion: database.example.org/v1alpha1
+    apiVersion: database.example.org/v1
     kind: XPostgreSQLInstance
   functions:
-  - name: my-cool-function
-    type: Container
-    # Configuration specific to `type: Container` functions.
-    container:
-      # The OCI image to pull and run.
-      image: xkpg.io/my-cool-function:0.1.0
-      # Whether to pull the function Never, Always, or IfNotPresent.
-      imagePullPolicy: IfNotPresent
-      # Secrets used to pull from a private registry.
-      imagePullSecrets:
-      - namespace: crossplane-system
-        name: my-xpkg-io-creds
-      # Note that only resource limits are supported - not requests.
-      # The function will be run with the specified resource limits.
-      resources:
-        limits:
-          memory: 64Mi
-          cpu: 250m
-      # Defaults to 'Isolated' - i.e an isolated network namespace.
-      network: Accessible
-      # How long the function may run before it's killed. Defaults to 10s.
-      timeout: 30s
-      # Containers are run by an external process listening at the supplied
-      # endpoint. Specifying an endpoint is optional; the endpoint defaults to
-      # the below value.
-      runner:
-        endpoint: unix:///@crossplane/fn/default.sock
-    # An x-kubernetes-embedded-resource RawExtension (i.e. an unschemafied
-    # Kubernetes resource). Passed to the function as the config block of its
-    # FunctionIO.
+  - name: compose-xr-using-go-templates
+    functionRef:
+      name: go-templates
     config:
-      apiVersion: database.example.org/v1alpha1
-      kind: Config
-      metadata:
-        name: cloudsql
-      spec:
-        version: POSTGRES_9_6
+      apiVersion: example.org/v1
+      kind: GoTemplate
+      source: Inline  # Or Remote, if you want to pull templates from git.
```

Comment:
good call to include Remote as an option - when these templates can stand alone, as opposed to inline in the config, it will probably be easier to use standard tooling to lint/test/etc. the template.